tests/virt_autotest/xen_guest_irqbalance.pm

```diff
@@ -0,0 +1,201 @@
+# SUSE's openQA tests
+
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: xen domU irqbalance test
+# Maintainer: Julie CAO <JCao@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use version_utils qw(is_sle);
+use set_config_as_glue;
+use virt_autotest::common;
+use virt_autotest::utils qw(is_kvm_host guest_is_sle wait_guest_online print_cmd_output_to_file);
+use virt_utils qw(upload_virt_logs remove_vm restore_downloaded_guests);
+
+our $vm_xml_save_dir = "/tmp/download_vm_xml";
+
+sub run_test {
+    my $self = shift;
+
+    return if is_kvm_host;
+
+    save_original_guests();
```

Comment:
@Julie-CAO  If we only change the guest xml setting  for those with nproc<4, then we can do save_original_guests and restore_original_guests for these changed only. This will be more efficient, right? In qam tests, this will be very preferred, because they have much more guests(~10) in a single job. HDYT?

Reply:
yes, that's true. I need create another function to handle the changed guests.